File: 0x06-python-classes/4-square.py
#!/usr/bin/python3
"""Class square create one object with an size
    size have verifications(type-data / value-data)

Raises:
    TypeError: [When size is not int]
    ValueError: [When is less that Zero]
    TypeError: [When value is not int]
    ValueError: [When value less that Zero]

Returns:
    [int] -- [size how public attribute and total area the object]
"""


class Square:
    """class constructor"""
    def __init__(self, size=0):
        if type(size) is not int:
            raise TypeError('size must be an integer')
        elif size < 0:
            raise ValueError('size must be >= 0')
        self.__size = size

    """size getter"""
    @property
    def size(self):
        return self.__size

    """size setter"""
    @size.setter
    def size(self, value):
        if type(value) is not int:
            raise TypeError('size must be an integer')
        elif value < 0:
            raise ValueError('size must be >= 0')
        self.__size = value

    """method for obtain area the object"""
    def area(self, __size=0):
        return self.__size * self.__size

File: 0x06-javascript-classes/4-square.js
#!/usr/bin/node
module.exports = class Square {
  constructor (size = 0) {
    try {
      if (!Number.isInteger(size)) throw 'size must be an integer';
      else if (size < 0) throw 'size must be >= 0';
      this._size = size;
    } catch (err) {
      console.log(err);
    }
  }

  area () {
    return this._size * this._size;
  }

  get size () {
    return this._size;
  }

  set size (value) {
    try {
      if (!Number.isInteger(value)) throw 'size must be an integer';
      else if (value < 0) throw 'size must be >= 0';
      this._size = value;
    } catch (err) {
      console.log(err);
    }
  }
};
